packages/core/src/provider/provider.ts

```diff
@@ -1,39 +1,17 @@
 import { EventEmitter } from "events";
-import { Query } from "../types";
+import { IWorld, Query } from "../types";
 
-interface IProvider {
-    set_entity(component: number,
-        query: Query,
-        offset: number,
-        value: number[],
-        calldata?: any[]): Promise<any>;
-    get_component(component: string, entity_id: string, offset: string, length: string): Promise<number>;
-    get_entity(entity_id: string): Promise<number>;
-    get_entities(entites: any[]): Promise<number[]>;
-}
-
-export abstract class Provider extends EventEmitter implements IProvider {
+export abstract class Provider extends EventEmitter implements IWorld {
     private readonly worldAddress: string;
 
     constructor(worldAddress: string) {
         super();
         this.worldAddress = worldAddress;
     }
 
-    // components
-    public abstract get_component(component: string, entity_id: string, offset: string, length: string): Promise<number>;
-
-    // entities
-    public abstract set_entity(component: number,
-        query: Query,
-        offset: number,
-        value: number[],
-        calldata?: any[]): Promise<any>;
-    public abstract get_entity(entity_id: string): Promise<any>;
-    public abstract get_entities(entites: any[]): Promise<any>;
-
+    public abstract entity(component: bigint, query: Query, offset: number, length: number): Promise<Array<bigint>>;
```

Comment:
i think on the client, we want an ability to query multiple components for an entity, so this api ideally looks more like the higher level commands api than the lower level world interface

Reply:
shouldn't we expose the world interface entirely in the provider? This was just going to be a 1:1 copy of the cairo interface